pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -209,10 +209,11 @@ func (w *PodWebhook) Default(ctx context.Context, obj runtime.Object) error {
 				return err
 			}
 		}
+
+		// copy back to the object
+		pod.pod.DeepCopyInto(obj.(*corev1.Pod))
```

Comment:
Moving the copy inside the that actually mutates the pod is a small cleanup / micro-optimization.  I could revert if we want to do it unconditionally (in case there is a future change that introduces another branch that also mutates the pod).

Reply:
Good catch, but I suggest a separate PR. We may consider backporting it to previous releases as a performance fix. Also, in case there is a bug - we can rollback easily just that.